server/test/lib/device/device.getDeviceFeaturesStates.test.js

```diff
@@ -0,0 +1,136 @@
+const EventEmitter = require('events');
+const { expect, assert } = require('chai');
+const uuid = require('uuid');
+const { fake } = require('sinon');
+const db = require('../../../models');
+const Device = require('../../../lib/device');
+
+const event = new EventEmitter();
+
+const now = new Date('2000-06-15T03:59:00.000Z');
+const insertStates = async () => {
+  const queryInterface = db.sequelize.getQueryInterface();
+  const deviceFeatureStateToInsert = [];
+  const statesToInsert = 3 * 60;
+  for (let i = 0; i < statesToInsert; i += 1) {
+    const startAt = new Date(now.getTime() - 3 * 60 * 60 * 1000);
+    const date = new Date(startAt.getTime() + ((3 * 60 * 60 * 1000) / statesToInsert) * i);
+    deviceFeatureStateToInsert.push({
+      id: uuid.v4(),
+      device_feature_id: 'ca91dfdf-55b2-4cf8-a58b-99c0fbf6f5e4',
+      value: i,
+      created_at: date,
+      updated_at: date,
+    });
+  }
+  await queryInterface.bulkInsert('t_device_feature_state', deviceFeatureStateToInsert);
+};
+
+describe('Device.getDeviceFeaturesStates', function Describe() {
+  this.timeout(15000);
+
+  beforeEach(async () => {
+    const queryInterface = db.sequelize.getQueryInterface();
+    await queryInterface.bulkDelete('t_device_feature_state');
+  });
+  it('Should return the full 24h existing state of the device feature - with queries "from" and "to" in GMT', async () => {
+    await insertStates();
+    const variable = {
+      getValue: fake.resolves(null),
+    };
+    const stateManager = {
+      get: fake.returns({
+        id: 'ca91dfdf-55b2-4cf8-a58b-99c0fbf6f5e4',
+        name: 'my-feature',
+      }),
+    };
+    const dateStateFrom = `${now.getUTCFullYear()}-${`0${now.getUTCMonth() + 1}`.slice(-2)}-${`0${now.getUTCDate() -
+      1}`.slice(-2)}`;
+    const dateStateTo = `${now.getUTCFullYear()}-${`0${now.getUTCMonth() + 1}`.slice(
```

Comment:
this will break when we'll reach months with 2 digits.

Why not use some static data for those tests?

Reply:
I found this formula on the net, it will never be the case because if 1 digit `now.getUTCMonth() + 1 = 1` then `(01).slice(-2) = 01` and if 2 digit ` now.getUTCMonth() + 1 = 10` then `(010).slice(-2) = 10`.
But you're right, you might as well register it in static. I modify that.

Reply:
Fixed